src/core/SampleClip.cpp

```diff
@@ -179,26 +179,22 @@ void SampleClip::setSampleBuffer(std::shared_ptr<const SampleBuffer> sb)
 
 void SampleClip::setSampleFile(const QString& sf)
 {
-	int length = 0;
-
 	if (!sf.isEmpty())
 	{
-		//Otherwise set it to the sample's length
 		m_sample = Sample(gui::SampleLoader::createBufferFromFile(sf));
-		length = sampleLength();
+		// Remove any prior offset in the clip
+		setStartTimeOffset(0);
+		updateLength();
 	}
-
-	if (length == 0)
+	else
 	{
-		//If there is no sample, make the clip a bar long
+		// If there is no sample, make the clip a bar long
 		float nom = Engine::getSong()->getTimeSigModel().getNumerator();
 		float den = Engine::getSong()->getTimeSigModel().getDenominator();
-		length = DefaultTicksPerBar * (nom / den);
+		changeLength(DefaultTicksPerBar * (nom / den));
+		setStartTimeOffset(0);
```

Comment:
`setStartTimeOffset(0)` is called in both branches. Can you move it out of both?